pvlib/bifacial/infinite_sheds.py

```diff
@@ -0,0 +1,796 @@
+    surface_azimuth : numeric
+        Surface azimuth angles in decimal degrees east of north
+        (e.g. North = 0, South=180 East = 90, West = 270). surface_azimuth must
+        be >=0 and <=360.
+    axis_azimuth : float or None, default None
+        The azimuth of the axis of rotation. Decimal degrees east of north.
+        For fixed tilt, set axis_azimuth = None.
+
+    Returns
+    -------
+    float or np.ndarray
+        Calculated rotation angle(s) in [deg]
+
+    Notes
+    -----
+    Based on pvfactors.geometry.base._get_rotation_from_tilt_azimuth
+    """
+    if axis_azimuth is None:
+        # Assume fixed tilt. Place axis_azimuth 90 degrees clockwise so that
+        # tilt becomes a negative rotation (lowers left/bottom edge)
+        axis_azimuth = ((surface_azimuth + 90.) + 360) % 360
+    # Calculate rotation of PV row (signed tilt angle)
+    is_pointing_right = ((surface_azimuth - axis_azimuth) % 360.) < 180.
+    rotation = np.where(is_pointing_right, surface_tilt, -surface_tilt)
+    return rotation
+
+
+def _vf_ground_sky_integ(surface_tilt, surface_azimuth, gcr, height,
+                         pitch, axis_azimuth=None, max_rows=5, npoints=100):
+    """
+    Integrated and per-point view factors from the ground to the sky at points
+    between interior rows of the array.
+
+    Parameters
+    ----------
+    surface_tilt : numeric
+        Surface tilt angle in degrees from horizontal, e.g., surface facing up
+        = 0, surface facing horizon = 90. [degree]
+    surface_azimuth : numeric
+        Surface azimuth angles in decimal degrees east of north
+        (e.g. North = 0, South=180 East = 90, West = 270). surface_azimuth must
+        be >=0 and <=360.
+    gcr : float
+        Ratio of row slant length to row spacing (pitch). [unitless]
+    height : float
+        Height of the center point of the row above the ground; must be in the
+        same units as ``pitch``.
+    pitch : float
+        Distance between two rows. Must be in the same units as ``height``.
+    axis_azimuth : numeric, default None
+        The compass direction of the axis of rotation lies for a single-axis
+        tracking system. Decimal degrees east of north.
+    max_rows : int, default 5
+        Maximum number of rows to consider in front and behind the current row.
+    npoints : int, default 100
+        Number of points used to discretize distance along the ground.
+
+    Returns
+    -------
+    fgnd_sky : float
+        Integration of view factor over the length between adjacent, interior
+        rows. [unitless]
+    fz : ndarray
+        Fraction of distance from the previous row to the next row. [unitless]
+    fz_sky : ndarray
+        View factors at discrete points between adjacent, interior rows.
+        [unitless]
+
+    """
+    z = np.linspace(0, 1, npoints)
+    rotation = np.atleast_1d(_tilt_to_rotation(
+        surface_tilt, surface_azimuth, axis_azimuth))
```

Comment:
Thanks for this. I agree that we can simplify here with `rotation=surface_tilt` because the ground discretization is nailed down to [0, 1] inside this private function, and these calculations do not allow for a tilted-axis single axis tracker. It's worth a comment in the code justifying the non-distinction between rotation and tilt, I'll take a stab at that.

Reply:
Oh wow, good point.  Is there a realistic use case for not having `max_rows` be large enough to at least closely approximate the asymptotic limit?  I imagine `max_rows=10` or `50` or something would be sufficient in any realistic scenario.

If the code can't be kept clean without `axis_azimuth` then I don't object to keeping it.  If `axis_azimuth` is only useful to be able to model scenarios like that exaggerated test case, I'm less convinced.

Reply:
We could remove `max_rows` from the parameter list of `infinite_sheds.get_irradiace_poa` and use a large internal limit such as 50. Doing so would make generating a known solution for testing `infinite_sheds._vf_ground_sky_integ` difficult, but not impractical.

The function `utils._ground_sky_vf_2d` honors its docstring, I think, and I would like to keep the `max_rows` parameter there.

If at some point these 2D models are extended to compute irradiance for rows close to the front (or rear) of the array, then `max_rows` and `surface_tilt <> rotation` would become issues to solve. That would be a reason to keep `axis_azimuth` for now. Users modeling fixed tilt arrays can ignore it (but may question why `axis_azimuth` is there), users modeling tracked systems already have to supply it.

Reply:
My inclination is to keep `axis_azimuth` in hopes that some future development extends this algorithm to compute irradiance for rows near or at the front of the array. I removed `max_rows` from the public functions (substituting an internal value of 100 x gcr, which ensures the view factor of the farthest row is < 0.005) in keeping with the theme that `infinite_sheds` computes irradiance in the middle of an array (no edge _or_ front/back row effects).

Reply:
We can always add a parameter later if/when this algorithm is extended, so I don't see much point in keeping a no-effect parameter in the meantime.  But I won't harp on it, whatever you want to do here is fine with me.